vmdb/app/controllers/ems_infra_controller.rb

```diff
@@ -17,6 +17,51 @@ def self.table_name
   def index
     redirect_to :action => 'show_list'
   end
+  
+  def scaling
+    assert_privileges("ems_infra_scale")
+    if params[:cancel]
+      redirect_to :action => 'show', :id => params[:id]
+    end
+    
+    drop_breadcrumb(:name => _("Scale Infrastructure Provider"), :url => "/ems_infra/scaling")
+    @infra = EmsOpenstackInfra.find(params[:id])
+    @stack = @infra.orchestration_stacks.first
+    @count_parameters = @stack.parameters.select {|x| x.name.to_s.include?('::count')}
+    
+    if !params[:scale]
+      return
+    end
+
+    scale_parameters = params.select { |k,v| k.include?('::count')}
+    
+    assigned_hosts = 0
+    scale_parameters.each_value { |value| assigned_hosts += Integer(value) }
+    infra = EmsOpenstackInfra.find(params[:id])
+    if assigned_hosts > infra.hosts.count
+      # Validate number of selected hosts is not more than available
+      add_flash(_("Assigning %{new} but only have %{old} hosts available." % {:new => assigned_hosts, :old => infra.hosts.count}), :error)
+      $log.error(_("Assigning %{new} but only have %{old} hosts available." % {:new => assigned_hosts, :old => infra.hosts.count}))
+    else
+      scale_parameters_formatted = []
+      return_message = "Scaling"
+      for p in @count_parameters
+        if !scale_parameters[p.name].nil? && scale_parameters[p.name] != p.value
+           return_message += " #{p.name} from #{p.value} to #{scale_parameters[p.name]}"
+           scale_parameters_formatted << {"name" => p.name, "value" => scale_parameters[p.name]}
+        end
+      end
+      if scale_parameters_formatted.length > 0
+        # A value was changed
+        stack = OrchestrationStackOpenstackInfra.find(params[:orchestration_stack_id])
```

Comment:
you already have this loaded in @stack, actually you should load stack this way, if you send thatp aram